.github/workflows/runtimes-digest-updater-upstream-v2.yaml

```diff
@@ -0,0 +1,102 @@
+---
+# The aim of this GitHub workflow is to update the runtimes ImageStreams
+name: Update runtime ImageStreams SHA digests
+on:  # yamllint disable-line rule:truthy
+  workflow_dispatch:
+    inputs:
+      branch:
+        required: true
+        description: "Which branch do you want to update?"
+      tag_version:
+        required: true
+        description: "Provide tag version: main or YYYYx"
+      user-hash:
+        required: false
+        description: "Optional: Specify a Git hash (it should exist in the provided branch's history)"
+
+env:
+  TMP_BRANCH: tmp-branch-${{ github.run_id }}
+  BRANCH_NAME: ${{ github.event.inputs.branch || 'main' }}
+  TAG_VERSION: ${{ github.event.inputs.tag_version || 'main' }}
+  USER_HASH: ${{ github.event.inputs.user-hash }}
+
+jobs:
+  initialize:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Install Skopeo CLI
+        shell: bash
+        run: |
+          sudo apt-get -y update
+          sudo apt-get -y install skopeo yq jq
+
+      - name: Checkout branch
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ env.BRANCH_NAME }}
+
+      - name: Create a new branch
+        run: |
+         echo ${{ env.TMP_BRANCH }}
+         git checkout -b ${{ env.TMP_BRANCH }}
+         git push --set-upstream origin ${{ env.TMP_BRANCH }}
+
+  update-runtimes:
+    needs: [initialize]
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Configure Git
+        run: |
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          git config --global user.name "GitHub Actions"
+
+      - name: Checkout release branch
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ env.TMP_BRANCH }}
+          fetch-depth: 0
+
+      - name: Invoke script to handle the update
+        shell: bash
+        run: |
+          chmod +x "${GITHUB_WORKSPACE}/ci/runtimes-digest-updater.sh"
```

Comment:
Do we need this step since this script is invoked via `bash ...` on the next line? And if we do, shouldn't we simply add these permissions to this script directly in our GH repo so we don't need to do it here everytime?